src/test/java/net/snowflake/client/jdbc/PreparedStatement2IT.java

```diff
@@ -337,103 +328,105 @@ public void testSnow35923() throws Exception {
   @Test
   @ConditionalIgnoreRule.ConditionalIgnore(condition = RunningOnGithubAction.class)
   public void testBindObjectLiteral() throws Exception {
-    try (Connection conn = init()) {
-      Statement stmt = conn.createStatement();
+    long t1Id = 0;
+    long t2Id = 0;
+
+    try (Connection conn = init();
+        Statement stmt = conn.createStatement()) {
 
       String sqlText = "create or replace table identifier(?) (c1 number)";
       SnowflakePreparedStatementV1 pStmt =
           (SnowflakePreparedStatementV1) conn.prepareStatement(sqlText);
       String t1 = "bindObjectTable1";
       // Bind the table name
       pStmt.setString(1, t1);
-      ResultSet result = pStmt.executeQuery();
+      try (ResultSet result = pStmt.executeQuery()) {}
 
       // Verify the table has been created and get the table ID
       stmt.execute("select parse_json(system$dict_id('table', '" + t1 + "')):entityId;");
-      result = stmt.getResultSet();
-
-      long t1Id = 0;
-      if (result.next()) {
-        t1Id = Long.valueOf(result.getString(1));
+      try (ResultSet result = stmt.getResultSet()) {
+        if (result.next()) {
+          t1Id = Long.valueOf(result.getString(1));
+        }
+        assertTrue(t1Id != 0);
       }
 
-      assertTrue(t1Id != 0);
-
       // Mix of object literal binds and value binds
       sqlText = "insert into identifier(?) values (1), (2), (3)";
       pStmt = (SnowflakePreparedStatementV1) conn.prepareStatement(sqlText);
       pStmt.setParameter("resolve_object_ids", true);
       // Bind by object IDs
       pStmt.setLong(1, t1Id);
 
-      result = pStmt.executeQuery();
+      try (ResultSet result = pStmt.executeQuery()) {}
 
       // Perform some selection
       sqlText = "select * from identifier(?) order by 1";
       pStmt = (SnowflakePreparedStatementV1) conn.prepareStatement(sqlText);
       pStmt.setString(1, t1);
-      result = pStmt.executeQuery();
-      // Verify 3 rows have been inserted
-      for (int i = 0; i < 3; i++) {
-        assertTrue(result.next());
+      try (ResultSet result = pStmt.executeQuery()) {
+        // Verify 3 rows have been inserted
+        for (int i = 0; i < 3; i++) {
+          assertTrue(result.next());
+        }
+        assertFalse(result.next());
       }
-      assertFalse(result.next());
 
       // Alter Table
       sqlText = "alter table identifier(?) add column c2 number";
       pStmt = (SnowflakePreparedStatementV1) conn.prepareStatement(sqlText);
       pStmt.setParameter("resolve_object_ids", true);
       pStmt.setLong(1, t1Id);
-      result = pStmt.executeQuery();
+      try (ResultSet result = pStmt.executeQuery()) {}
 
       // Describe
       sqlText = "desc table identifier(?)";
       pStmt = (SnowflakePreparedStatementV1) conn.prepareStatement(sqlText);
```

Comment:
prepared statement should also be in try with resources - please take also look at them in the other changed tests

Reply:
Done. I wrapped up all the pstmt with the try with resources in this file.